chrome_extension/background.js

```diff
@@ -19,21 +19,31 @@
 const GREEN_ICON_PATH = './icon-green.png';
 const RED_ICON_PATH = './icon-red.png';
 
-const isUsingDesiredCspPerTab = new Map();
+// Map of tabs that are applying the desired CSP. The key is the `tabId`, the
+// value is the specific `url` that applies them. Keeping track of the URL is
+// necessary since administrators can block access to the `chrome.webRequest`
+// API used to check CSP on a per page basis. Without checking that the URL
+// matches, a user could navigate from a secure page to a potentially
```

Comment:
How does the user navigate to a different page?

Is it only if the user types a URL in the address bar, or can a user click on a link to also navigate or is this blocked by the CSP?

Reply:
For this case form of navigation works, clicking a bookmark, typing a URL etc. As long as it is in the same tab.

Reply:
RE disabling in page navigation, it prevents some, unfortunately not all. I created an issue to tackle this: https://github.com/project-oak/oak/issues/1792